src/sliceDataStorage.cpp

```diff
@@ -287,16 +315,17 @@ Polygons
             }
             else
             {
-                return raftOutline;
+                return *raftOutline;
             }
         }
         else
         {
             return Polygons();
         }
+        break;
     }
-    else
-    {
+    case Raft::LayerType::Airgap:
+    case Raft::LayerType::Model:
         Polygons total;
         if (layer_nr >= 0)
         {
```

Comment:
Compiler gives me a `control reaches end of non-void function` warning because there is no `default` in your `switch`. This should definitely not happen, but I suggest returning an empty value after the switch.

Reply:
But the switch is exaustive? hmm maybe some weird cpp quircks. I'' fix it

Reply:
https://github.com/Ultimaker/CuraEngine/pull/2006/commits/68d0de9d4dab7ef05683d21c0514a2c1158e2af9

Reply:
Yes it is, but it is technically possible to put a integer value in your variable that is not a valid enum value, in which case it will go out the switch, having no `default`. In this case you have no return value, which is very bad :) So the compiler issues a warning.